Source/RapyutaSimulationPlugins/Private/Robots/RRBaseRobot.cpp

```diff
@@ -28,9 +29,37 @@ void ARRBaseRobot::SetupDefault()
     // Generally, for sake of dynamic robot type import/creation, child components would be then created on the fly!
     // Besides, a default subobject, upon content changes, also makes the owning actor become vulnerable since one in child BP actor
     // classes will automatically get invalidated.
+    URRUObjectUtils::SetupDefaultRootComponent(this);
 
-    RootComponent = CreateDefaultSubobject<USceneComponent>(TEXT("Root"));
-    ROS2InterfaceClass = URRRobotROS2Interface::StaticClass();

```

Comment:
why is the default initialization of `ROS2InterfaceClass` removed?

Reply:
Yeah, it is noted [here](https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/SootballUtils2/Source/RapyutaSimulationPlugins/Private/Robots/RRBaseRobot.cpp#L34).
I also don't actually favor leaving it null here, but since child BP class may reconfig this class, creating another instance in `PostInitializeComponents()`.
So basically, `ARRBaseRobot` should not be instantiated directly. This is actually very similar to `ARRBaseRobotVehicle::VehicleComponentClass`, which is reconfigured by child BPs classes like `BP_TurtlebotBurgerPhysics/BP_TurtlebotWaffleVehicle`